src/components/axes/timeAxis.ts

```diff
@@ -306,10 +320,20 @@ export module Axis {
         );
     }
 
-    private _cleanTier(index: number) {
-      this._tierLabelContainers[index].selectAll("." + AbstractAxis.TICK_LABEL_CLASS).remove();
-      this._tierMarkContainers[index].selectAll("." + AbstractAxis.TICK_MARK_CLASS).remove();
-      this._tierBaselines[index].style("visibility", "hidden");
+    private _cleanTiers() {
+      var oldTiersNumber = this._tierLabelContainers.length;
+
+      if (! (this._tierLabelContainers.length === this._tierMarkContainers.length &&
+             this._tierLabelContainers.length === this._tierBaselines.length)) {
+        throw Error("Tiers could not be cleaned properly");
+      }
```

Comment:
It should never occur. It is just defensive code (which I believe we are missing a lot in the codebase). I can remove it if you want. This is just because the method (and what was here before) assumes that the three arrays are having the same length. 

What is our policy about defensive code like this. It would be great to have some Java-like assert system